clang/include/clang/Tooling/DependencyScanning/ModuleDepCollector.h

```diff
@@ -138,6 +139,14 @@ struct ModuleDeps {
   /// determined that the differences are benign for this compilation.
   std::vector<ModuleID> ClangModuleDeps;
 
+  /// The set of libraries or frameworks to link against when
+  /// an entity from this module is used.
+  llvm::SmallVector<Module::LinkLibrary, 2> LinkLibraries;
+
+  /// Autolinking uses the framework name for linking purposes
+  /// when this is false and the export_as name otherwise.
+  bool UseExportAsModuleLinkName;
```

Comment:
How would this be used externally? We aren't providing export_as, nor identifying frameworks vs non-frameworks. Incidentally, you aren't exposing this in clang-scan-deps output as far as I can see.

If the idea is to drop the autolink of the current module, should we rework that so that we remove it from the link libraries of the importer instead?  It's a weird concept here since you have already resolved all the dependencies.

Reply:
Right, the question in my mind is if we should model that in the scanner or change the modulemap code to drop the LinkLibraries.  I don't know why it was originally modeled as a flag handled in codegen instead of changing the LinkLibraries.

Reply:
It seems that `LinkLibraries` are still collected for a `Module` separately just in case it gets discovered/imported by a module matching a `export_as` name. So I can see why a separate flag was introduced. 

The logic for adding linker flags in CodeGen omits these modules if this flag (`UseExportAsModuleLinkName`) is set, and that matching the intent behind this change here, I think it would make sense to model the same in the scanner. i.e. omit `LinkLibraries` of such module, because by the time we've discovered all the dependencies we are guaranteed to have the required knowledge to make this decision.